### ISSUE TITLE
Page/Asset API support in builder (enhancement)

### ISSUE THREAD

**Issue by 3xil3** [valtech/aem-easy-content-upgrade#21]
Working resource based definitely covers the basics, but for some more advanced migrations, you could benefit from some additional extensions to the builder (or a different builder could be used)

For example:
Moving specific pages usually involves also updating pages referencing the 'moved' page as otherwise you end up with broken links.

Simular use case for assets: when moving an asset, you usually want to update all references in pages (or other resources) using that asset.

**Comment by gruberrolandvaltech** [comment]
True, we will think about how to deal with the reference updates.
Page actions moved to separate ticket: https://github.com/valtech/aem-easy-content-upgrade/issues/23

**Comment by gruberrolandvaltech** [comment]
new actions required:
.doReplaceValueInProperty(old, new)
.doReplaceValueInProperty(old, new, String[] propertyNames)
.doReplaceValueInPropertyRegex(searchRegex, replacement)
.doReplaceValueInPropertyRegex(searchRegex, replacement, String[] propertyNames)

Example: .doReplaceValueInPropertyRegex("/content/dam/(project[0-9]+)", "/content/dam/projects/$1")

The properties can be single and multi valued.
